src/solvers/td3.ml

```diff
@@ -297,10 +308,26 @@ module WP =
         delete_marked wpoint;
         delete_marked stable;
 
-        print_data data "Data after clean-up"
+
+        print_data data "Data after clean-up";
+
+        List.iter set_start st;
+
+        if not !any_changed_start_state && GobConfig.get_bool "incremental.reluctant" then (
+          (* solve on the return node of changed functions. Only destabilize influenced nodes outside the function if the analysis result changed *)
+          print_endline "Separately solving changed functions...";
+          Hashtbl.iter (fun x (old_rho, old_infl) -> ignore @@ Pretty.printf "test for %a\n" Node.pretty_trace (S.Var.node x);
+            solve x Widen;
+            if not (S.Dom.leq (HM.find rho x) old_rho) then (
```

Comment:
The downside here is also that if the change makes the state more precise, the new precision gets discarded in favor of the old less precise solution. This is a problem if we want to allow the user to fix a problem and have the related warnings disappear without having to reanalyze from scratch.

Reply:
Maybe I should create a configuration option to specify whether to use `leq` or `eq` at this point? I think both can have their benefits. `eq` is useful if one does want more precision, whereas `leq` might be the better choice when one focuses on reducing the number of evaluations and possibly the runtime of the analysis.

Reply:
Having an option might be good indeed.